### PR TITLE
Fix #839: Added URP cookie logic

### LINKED ISSUE #839: Add Stored Cookies for URP Partners
Certain partners need to use stored cookies verse HTTP Headers.

### PR DESCRIPTION
**Pull request by danishjafri88**
## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-ios/issues) for my issue if one did not already exist.
- [x] My patch or PR title has a standard commit message that looks like `Fix #123: This fixes the shattered coffee cup!` (or `No Bug: <message>` if no relevant ticket)
- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`
- [x] New files have MPL-2.0 license header.

## Reviewer Checklist:

- [x] PR is linked to an issue via [Zenhub](https://www.zenhub.com/extension).
- [x] Issues are assigned to at least one epic.
- [x] Issues include necessary QA labels:
  - [x] `QA/(Yes|No)`
  - [x] `release-notes/(include|exclude)`
  - [x] `bug` / `enhancement`
- [x] Necessary security reviews have taken place.
- [x] Adequate test coverage exists to prevent regressions.
- [x] Adaquate test plan exists for QA to validate (if applicable)

### CODE REVIEW

**Comment by jumde** [comment]
PR lgtm, its only missing the `__Secure-` prefix as discussed here: https://github.com/brave/brave-ios/pull/818/files/c7fcd68f5c319bf2fcc7925e4e7bb6603a682f1f#r252019733  cc: @danishjafri88 @jhreis

**Comment by jumde** [comment]
++

**Comment by diracdeltas** [comment]
@jumde is this still missing the httpOnly flag? sorry if i missed it

**Comment by jumde** [comment]
@diracdeltas - The flag is read-only https://github.com/brave/brave-ios/pull/818/files#r252353535

**Comment by diracdeltas** [comment]
i think you can use this method to initialize a cookie with the HttpOnly flag: https://developer.apple.com/documentation/foundation/httpcookie/1393011-cookies

**Comment by jhreis** [comment]
@diracdeltas this function creates a non-configured array of cookies (just header key/value pairs). 
There is a [cookie constructor](https://developer.apple.com/documentation/foundation/httpcookie/1392975-init) (used [here](https://github.com/brave/brave-ios/pull/823/files#diff-197f5cf766a44b6162c4c8fe04cbc865R32)) that allows us to configure most of the `readonly` properties (e.g. `secure`), but `httponly` is not one of the options we can even use there. Here is a list of the [constructor configurable properties](https://developer.apple.com/documentation/foundation/httpcookiepropertykey).

**Comment by diracdeltas** [comment]
@jhreis from https://medium.com/@mahithegeek/cookies-in-ios-nshttpcookie-and-nshttpcookiestorage-696564bc6975 (method 2) it seems like there exists a way to construct a cookie from an HTTP response string which could set the httponly flag, but i probably linked the wrong docs in my previous comment because i am an iOS n00b

File: BraveShared/Analytics/UserReferralProgram.swift
Comment (iccub) [inline BraveShared/Analytics/UserReferralProgram.swift]: `: [String]` can be removed.

File: Client/Frontend/Browser/TabManager.swift
Comment (iccub) [inline Client/Frontend/Browser/TabManager.swift]: `self` can be removed

File: Client/Frontend/Browser/TabManager.swift
Comment (iccub) [inline Client/Frontend/Browser/TabManager.swift]: `self` can be removed

**Review by iccub — COMMENTED** [review]
Looks good, only few nitpicks to remove

File: Client/Frontend/Browser/TabManager.swift
Comment (danishjafri88) [reply Client/Frontend/Browser/TabManager.swift]: removed :)

File: Client/Frontend/Browser/TabManager.swift
Comment (danishjafri88) [reply Client/Frontend/Browser/TabManager.swift]: removed

**Review by iccub — APPROVED** [review]
Approving and waiting for @jhreis for final approval and merge.

**Comment by jumde** [comment]
Disabling HTTPOnly flag is fine for this PR. cc: @diracdeltas. Detailed conversation here: https://bravesoftware.slack.com/archives/G9Q8HHBQE/p1549045221000300

**Pull request merged**
